agent/allow_list.go

```diff
@@ -0,0 +1,81 @@
+package agent
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type allowList struct {
+	Exec      string   `yaml:"exec" json:"exec"`
+	Args      []string `yaml:"args" json:"args"`
+	Sha512    string   `yaml:"sha512" json:"sha512"`
+	EnableEnv bool     `yaml:"enable_env" json:"enable_env"`
+}
+
+func readAllowList(path string, readBytes func(string) ([]byte, error)) ([]allowList, error) {
+	var allowList []allowList
+	if path == "" {
+		return allowList, nil
+	}
+	unmarshalFuncs := map[string]func(in []byte, out interface{}) error{
```

Comment:
Can we just use yaml unmarshaling in all cases? JSON is a subset of YAML. Then we wouldn't need to care about the extension.

Reply:
yaml.Unmarshal works on JSON files?

Reply:
When using `yaml.Unmarshal` in all cases, I'm getting some errors: `yaml: line 2: found character that cannot start any token`. I believe this is due to the use of tabs in JSON rather than spaces. I think we may still want to support tabs, wdyt?

Reply:
I see. Well don't worry about it then :)